src/metabase/public_settings.clj

```diff
@@ -205,38 +217,67 @@
-   :embedding             (enable-embedding)
-   :enable_nested_queries (enable-nested-queries)
-   :enable_query_caching  (enable-query-caching)
-   :enable_xrays          (enable-xrays)
-   :engines               (driver.u/available-drivers-info)
-   :entities              (types/types->parents :entity/*)
-   :ga_code               "UA-60817802-1"
-   :google_auth_client_id (resolve-setting 'metabase.api.session 'google-auth-client-id)
-   :has_sample_dataset    (db/exists? 'Database, :is_sample true)
-   :hide_embed_branding   (metastore/hide-embed-branding?)
-   :ldap_configured       (resolve-setting 'metabase.integrations.ldap 'ldap-configured?)
-   :map_tile_server_url   (map-tile-server-url)
-   :metastore_url         metastore/store-url
-   :password_complexity   password/active-password-complexity
-   :premium_token         (metastore/premium-embedding-token)
-   :public_sharing        (enable-public-sharing)
-   :report_timezone       (resolve-setting 'metabase.driver 'report-timezone)
-   :setup_token           (resolve-setting 'metabase.setup 'token-value)
-   :site_name             (site-name)
-   :site_url              (site-url)
-   :timezone_short        (short-timezone-name (setting/get :report-timezone))
-   :timezones             common/timezones
-   :types                 (types/types->parents :type/*)
-   :version               config/mb-version-info})
+(defsetting available-locales
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (available-locales-with-names)))
+
+(defsetting email-configured
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.email 'email-configured?)))
+
+(defsetting engines
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (driver.u/available-drivers-info)))
+
+(defsetting entities
+  (deferred-tru "TODO")
+  :visibility :authenticated
+  :getter     (fn [] (types/types->parents :entity/*)))
+
+(defsetting has-sample-dataset
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (db/exists? 'Database, :is_sample true)))
+
+(defsetting hide-embed-branding
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (metastore/hide-embed-branding?)))
+
+(defsetting ldap-configured
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.integrations.ldap 'ldap-configured?)))
+
+(defsetting password-complexity
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] password/active-password-complexity))
+
+(defsetting setup-token
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (resolve-setting 'metabase.setup 'token-value)))
+
+(defsetting timezone-short
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] (short-timezone-name (setting/get :report-timezone))))
+
+(defsetting timezones
+  (deferred-tru "TODO")
+  :visibility :public
+  :getter     (fn [] common/timezones))
```

Comment:
You can't set some of these virtual settings like `:timezones` so they need to have a `:setter` to reflect that. I think the easiest option is to allow `:setter` to be either a function or `:none`. The schema would be 

```clj
(s/cond-pre (s/eq :none) clojure.lang.IFn)
```

Then in whatever function gets the setting (I think it's `metabase.models.setting/set!` if `(= setter :none)` then it should throw an Exception. Something like this is probably fine:

```clj
(when (= setter :none)
  (throw (UnsupportedOperationException. (tru "You cannot set %s; it is a read-only setting." setting-name))))
```